### PR TITLE
[LPT] LP Transformations refactoring after dynamic shapes support

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details:
 - `get_output_shape`/`get_input_shape` removed from LPT
 - Clean-up transformations improvement: fusing per-tensor dequantization to FQ in dynamic rank cases
 - Some transformations refactoring

### Tickets:
 - *58782*

### Validation:
 - 62037

### CODE REVIEW

File: inference-engine/src/low_precision_transformations/src/fake_quantize.cpp [hunk 1]
```diff
@@ -56,8 +56,11 @@ bool FakeQuantizeTransformation::transform(TransformationContext& context, ngrap
 namespace fq {
 
 static std::shared_ptr<Node> updateShape(std::shared_ptr<Node> constantOp, const PartialShape& targetShape) {
-    const Shape shape = constantOp->get_output_shape(0);
-    if ((shape.size() < static_cast<size_t>(targetShape.rank().get_length())) && (shape.size() > 1ul)) {
+    const auto constant = as_type_ptr<opset1::Constant>(constantOp);
+    assert(constant != nullptr);
+
+    const Shape shape = constant->get_shape();
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/fake_quantize.cpp:62]: Do we really need to cast to Constant to get operation shape?

**Review by eshoguli — COMMENTED** [review]
nothing critical,  some questions about shape routines

File: inference-engine/src/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 2]
```diff
@@ -40,8 +40,13 @@ bool FuseFakeQuantizeTransformation::transform(TransformationContext& context, n
 
 namespace fuse_fq {
 
-std::shared_ptr<Node> updateShape(std::shared_ptr<Node> op, const Shape& targetShape) {
-    const Shape shape = op->get_output_shape(0);
+std::shared_ptr<Node> updateShape(std::shared_ptr<Node> op, const PartialShape& targetPShape) {
+    const auto constant = as_type_ptr<opset1::Constant>(op);
+    const Shape shape = constant->get_shape();
+
+    assert(targetPShape.is_static());
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/fuse_fake_quantize.cpp:44]: Yes, we can. Fixed, thanks!

File: inference-engine/src/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 3]
```diff
@@ -81,9 +86,12 @@ bool eltwiseWithConstant(const std::shared_ptr<Node>& eltwise) {
         return false;
     }
 
-    Shape shape = constant->get_output_shape(0);
+    Shape shape = constant->get_shape();
     if ((!shape.empty()) && (shape_size(shape) != 1ul)) {
-        const Shape eltwiseShape = eltwise->get_output_shape(0);
+        const auto eltwisePShape = eltwise->get_output_partial_shape(0);
+        assert(eltwisePShape.is_static());
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/fuse_fake_quantize.cpp]: Fixed, thanks!

File: inference-engine/src/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 4]
```diff
@@ -81,9 +86,12 @@ bool eltwiseWithConstant(const std::shared_ptr<Node>& eltwise) {
...
+        const Shape eltwiseShape = eltwisePShape.to_shape();
+
         if ((eltwiseShape.size() - shape.size()) > 1) {
             return false;
         }
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/fuse_fake_quantize.cpp]: Done, thanks

File: inference-engine/src/low_precision_transformations/src/transpose.cpp [hunk 5]
```diff
@@ -42,47 +42,39 @@ void transposeDequantizationConstant(std::shared_ptr<Node>& transpose) {
...
+            const auto constantShape = dequantizationConstant->get_shape();
+            if (shape_size(constantShape) == 1ul) {
+                return NetworkHelper::toScalar(dequantizationConstant);
             }
 
-            if (dequantizationShape.size() != static_cast<size_t>(transposeOutputShape.rank().get_length())) {
-                dequantizationConstant = fold<opset1::Unsqueeze>(
-                    dequantizationConstant,
-                    std::make_shared<opset1::Constant>(element::i32, Shape{ 1 }, std::vector<size_t>{0}));
+            const size_t transposeOutRank = transposeOutputShape.rank().get_length();
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/transpose.cpp]: These checks passed in `TransposeTransformation::canBeTransformed`. Added assertion

File: inference-engine/src/low_precision_transformations/src/fake_quantize.cpp (hunk 1 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/fake_quantize.cpp]: Fixed, thanks!

File: inference-engine/src/low_precision_transformations/src/convolution.cpp [hunk 6]
```diff
@@ -268,7 +266,10 @@ bool ConvolutionTransformation::transform(TransformationContext &context, ngraph
             } else {
                 subtractFromWeights = as_type_ptr<opset1::Subtract>(optimizedSubtract);
 
-                const Shape weightsShape = subtractFromWeights->input(0).get_shape();
+                const auto weightsPShape = subtractFromWeights->get_input_partial_shape(0);
+                assert(weightsPShape.is_static());
+                const Shape weightsShape = weightsPShape.to_shape();
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/convolution.cpp]: Yes, we can. Done, thanks

File: inference-engine/src/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 7]
```diff
@@ -40,8 +40,13 @@ bool FuseFakeQuantizeTransformation::transform(TransformationContext& context, n
 
 namespace fuse_fq {
 
-std::shared_ptr<Node> updateShape(std::shared_ptr<Node> op, const Shape& targetShape) {
-    const Shape shape = op->get_output_shape(0);
+std::shared_ptr<Node> updateShape(std::shared_ptr<Node> op, const PartialShape& targetPShape) {
+    const auto constant = as_type_ptr<opset1::Constant>(op);
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/fuse_fake_quantize.cpp]: Fixed, thank you!

**Review by eshoguli — APPROVED** [review]

**Pull request merged**